src/service/viewport-impl.js

```diff
@@ -1093,22 +1140,41 @@ export class ViewportBindingNaturalIosEmbed_ {
   }
 
   /** @override */
-  updatePaddingTop(paddingTop, opt_updateScrollPos, opt_lastPaddingTop) {
+  hideViewerHeader(transient, lastPaddingTop) {
+    onDocumentReady(this.win.document, doc => {
+      if (!transient) {
+        this.paddingTop_ = 0;
+        doc.body.style.paddingTop = '';
+        doc.body.style.borderTop = '';
+      } else {
+        // Add extra paddingTop to make the content stay at the same position
+        // when the hiding header operation is transient
+        const existingPaddingTop =
+            this.win./*OK*/getComputedStyle(doc.body)['padding-top'] || '0';
+        doc.body.style.paddingTop =
+            `calc(${existingPaddingTop} + ${lastPaddingTop}px)`;
+        doc.body.style.borderTop = '';
+      }
+    });
+  }
+
+  /** @override */
+  showViewerHeader(transient, paddingTop) {
+    onDocumentReady(this.win.document, doc => {
+      if (!transient) {
+        this.paddingTop_ = paddingTop;
```

Comment:
done.
I changed the showViewerHeader to do nothing when `transient==true` because we don't have to change borderTop when it's transient. Otherwise, the content will jump.